packages/vanchor-client/src/queries/deposit/depositDayInterval.ts

```diff
@@ -0,0 +1,261 @@
+import { execute } from '../../../.graphclient';
+import { DateUtil, getEpochArray } from '../../utils/date';
+import { SubgraphUrl } from '../../config';
+
+export interface DepositByChainDayIntervalItem {
+  subgraphUrl: SubgraphUrl;
+  deposit: number;
+  date: Date;
+  vAnchorAddress: string;
+}
+
+export interface DepositByChainAndByTokenDayIntervalItem
+  extends DepositByChainDayIntervalItem {
+  tokenSymbol: string;
+}
+
+export interface DepositByVAnchorDayIntervalItem {
+  vAnchorAddress: string;
+  deposit: number;
+}
+
+export interface DepositVAnchorsDateRangeItem {
+  [epoch: string]: number;
+}
+
+export const GetVAnchorDepositByChainDayInterval = async (
+  subgraphUrl: SubgraphUrl,
+  vAnchorAddress: string,
+  date: Date
+): Promise<DepositByChainDayIntervalItem> => {
+  const query = /* GraphQL */ `
+    query TotalValueLocked {
+      vanchorDepositEveryDays(
+        where: {
+          date: "${DateUtil.fromDateToEpoch(date)}",
+          vAnchorAddress: "${vAnchorAddress.toLowerCase()}"
+        }
+      ) {
+        deposit
+        vAnchorAddress
+        date
+      }
+    }
+  `;
+  const result = await execute(
+    query,
+    {},
+    {
+      subgraphUrl,
+    }
+  );
+
+  return result.data.vanchorDepositEveryDays?.map((item: any) => {
+    return {
+      deposit: +item?.deposit,
+      subgraphUrl: subgraphUrl,
+      date: DateUtil.fromEpochToDate(parseInt(item?.date)),
+      vAnchorAddress: item?.vAnchorAddress,
+    };
+  });
+};
+
+export const GetVAnchorDepositByChainsDayInterval = async (
+  subgraphUrls: Array<SubgraphUrl>,
+  vAnchorAddress: string,
+  date: Date
+): Promise<Array<DepositByChainDayIntervalItem>> => {
+  const promises: Array<Promise<DepositByChainDayIntervalItem>> = [];
+
+  for (const subgraphUrl of subgraphUrls) {
+    promises.push(
+      GetVAnchorDepositByChainDayInterval(subgraphUrl, vAnchorAddress, date)
+    );
+  }
+
+  return await Promise.all(promises);
+};
+
+export const GetVAnchorsDepositByChainDayInterval = async (
+  subgraphUrl: SubgraphUrl,
+  vanchorAddresses: Array<string>,
+  date: Date
+): Promise<Array<DepositByVAnchorDayIntervalItem>> => {
+  const query = /* GraphQL */ `
+    query TotalValueLockedByVAnchor {
+      vanchorDepositEveryDays(
```

Comment:
This is a weird name, can you propose or come up with a better name?
- `vanchorDepositDayData`
- `vanchorDepositsByDay`

wdyt @vutuanlinh2k2 @vutuanlinh2k2